src/GraphQL.Tests/TestExtensions.cs

```diff
@@ -1,19 +1,66 @@
 using System;
+using System.Collections;
 using System.Collections.Generic;
 using System.Linq;
+using GraphQL.Execution;
 
 namespace GraphQL.Tests
 {
     internal static class TestExtensions
     {
-        public static Dictionary<string, object> ToDict(this object data)
+        public static IReadOnlyDictionary<string, object> ToDict(this object data)
         {
             if (data == null)
                 return new Dictionary<string, object>();
 
-            return data is ObjectProperty[] properties
-                ? properties.ToDictionary(x => x.Key, x => x.Value)
-                : throw new ArgumentException($"Unknown type {data.GetType()}. Parameter must be of type ObjectProperty[].", nameof(data));
+            if (data is ObjectExecutionNode objectExecutionNode)
+                return (IReadOnlyDictionary<string, object>)objectExecutionNode.ToValue();
+
+            if (data is IReadOnlyDictionary<string, object> properties)
+            {
+                return properties;
+            }
+
+            throw new ArgumentException($"Unknown type {data.GetType()}. Parameter must be of type ObjectExecutionNode or IDictionary<string, object>.", nameof(data));
+        }
+
+        public static RootExecutionNode ToData(this IReadOnlyDictionary<string, object> dictionary)
+        {
+            var root = new RootExecutionNode(null)
+            {
+                SubFields = dictionary.Select(x => CreateExecutionNode(x.Key, x.Value)).ToArray()
+            };
+            return root;
+        }
+
+        private static ExecutionNode CreateExecutionNode(string name, object value)
+        {
+            if (value is IEnumerable<KeyValuePair<string, object>> dict)
+            {
+                return new ObjectExecutionNode(null, null, new GraphQL.Language.AST.Field(new GraphQL.Language.AST.NameNode(name), default), null, default)
+                {
+                    SubFields = dict.Select(x => CreateExecutionNode(x.Key, x.Value)).ToArray(),
+                };
+            }
+            else if (value?.GetType() != typeof(string) && value is IEnumerable list)
```

Comment:
What if value is `byte[]` ?

Reply:
Byte arrays are not native JSON types. So it should work correctly. 

I wrote the code to mimic the behavior in ObjectExtensions.GetPropertyValue and TypeExtensions.IsAnIEnumerable and AutoRegisteringGraphType.GetRealType, and so on. They all just check for a string and continue otherwise.